### PR TITLE
Implement opcode `RETURNDATACOPY`

### LINKED ISSUE #44: Opcode: RETURNDATACOPY

### PR DESCRIPTION
**Pull request by JulianVentura**
This PR implements the opcode `RETURNDATACOPY` with tests.

Depends on #322 #327

Closes #44

### CODE REVIEW

File: src/codegen/operations.rs [hunk 1]
```diff
@@ -5064,3 +5065,113 @@ fn codegen_returndatasize<'c, 'r>(
...
+    let end_block = region.append_block(Block::new(&[]));
+    let return_data_size = op_ctx.get_return_data_size(&ext_mem_block, location)?;
+    let return_data_size = ext_mem_block
+        .append_operation(arith::trunci(return_data_size, uint32.into(), location))
+        .result(0)?
+        .into();
+    let req_mem_size_ok = ext_mem_block
+        .append_operation(arith::cmpi(
+            context,
+            arith::CmpiPredicate::Sle,
```
Comment (MegaRedHand) [inline src/codegen/operations.rs:5156]: Why is this signed?

File: src/codegen/operations.rs [hunk 2]
```diff
@@ -5064,3 +5065,113 @@ fn codegen_returndatasize<'c, 'r>(
...
+        region,
+        req_mem_size,
+        gas_cost::RETURNDATACOPY,
+    )?;
+
+    //Check that offset + size < return_data_size
+    let end_block = region.append_block(Block::new(&[]));
+    let return_data_size = op_ctx.get_return_data_size(&ext_mem_block, location)?;
+    let return_data_size = ext_mem_block
+        .append_operation(arith::trunci(return_data_size, uint32.into(), location))
```
Comment (MegaRedHand) [inline src/codegen/operations.rs:5150]: Shouldn't we truncate `req_return_data_size` to `u64` instead?

File: src/syscall.rs [hunk 3]
```diff
@@ -391,33 +406,45 @@ impl<'c> SyscallContext<'c> {
...
-        // Check if the offset is within the target slice
-        if target_offset >= target.len() {
+        // Check if the offsets are within their respective slices
+        if size + target_offset > target.len() {
             eprintln!("ERROR: Specified target offset is bigger than target len");
             return;
         }
 
+        if source_offset > source.len() {
+            // Nothing to copy
```
Comment (MegaRedHand) [inline src/syscall.rs:436]: We should add a print here too.

File: src/codegen/operations.rs (hunk 1 above)
Comment (JulianVentura) [reply src/codegen/operations.rs:5156]: Done

File: src/syscall.rs (hunk 3 above)
Comment (JulianVentura) [reply src/syscall.rs:436]: Done

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**